src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxTreeFactoryService.cs

```diff
@@ -48,8 +48,15 @@ public override SyntaxNode DeserializeNodeFrom(Stream stream, CancellationToken
                 return CSharpSyntaxNode.DeserializeFrom(stream, cancellationToken);
             }
 
+            public override bool CanCreateRecoverableTree(SyntaxNode root)
+            {
+                var cu = root as CompilationUnitSyntax;
+                return base.CanCreateRecoverableTree(root) && cu != null && cu.AttributeLists.Count == 0;
```

Comment:
The attribute list check is a currently a mystery.  I retained it when I refactored the code, but the only one that seems to have known why it is there is Kyle, as this change is one of the last things he did before he left.